### PR TITLE
Fix for creating searchbar pages data - escaping apostrophes

### PR DESCRIPTION
**Pull request by pikinier20**

### CODE REVIEW

File: plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt [hunk 1]
```diff
@@ -107,3 +113,94 @@ class SourcesetDependencyAppender(val context: DokkaContext) : PageTransformer {
         )
     }
 }
+
+class SearchbarDataInstaller(val context: DokkaContext): PageTransformer {
+    private val pageList = mutableMapOf<String, Pair<String, String>>()
+    lateinit var locationProvider : LocationProvider
+
+        override fun invoke(input: RootPageNode): RootPageNode {
+            locationProvider = context.plugin<DokkaBase>().querySingle { locationProviderFactory }
```
Comment (MarcinAman) [inline plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt:122]: I am not so sure about fetching location provider like that. It requires a root page that can be changed in other `PageTransformer`, so unless you are entirely sure that this page transformer is last i'd just create this object in HtmlRenderer and pass location provider via constructor

File: plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt [hunk 2]
```diff
@@ -107,3 +113,94 @@ class SourcesetDependencyAppender(val context: DokkaContext) : PageTransformer {
...
+            )
+
+            return input.modified(
+                children = input.children + data
+            )
+    }
+
+    private fun traverse(page:PageNode): Unit {
+        if(page is ContentPage && page !is ModulePageNode && page !is PackagePageNode) processPage(page)
+        page.children.forEach(::traverse)
```
Comment (MarcinAman) [inline plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt:138]: [Minor] Imho a tailrec function with accumulator would look better than side effect

File: plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt (hunk 1 above)
Comment (pikinier20) [reply plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt]: I've changed usage of SearchbarDataInstaller - I don't declare it as PageTransformer and use it in HtmlRenderer to register and finally to generate page list.

File: plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt (hunk 2 above)
Comment (pikinier20) [reply plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt]: I deleted this recursion as it was failing with strange exceptions when tried to run it after rendering

**Pull request merged**